fluentd/config.sls

```diff
@@ -10,6 +10,15 @@ add_fluent_{{ config.name }}_config:
         - service: reload_fluentd_service
 {% endfor %}
 
+{% for name, path in salt.pillar.get('fluentd:persistent_directories', {}).items() %}
+create_directory_for_{{ name }}_logs:
+  file.directory:
+   - nane: {{ path }}
+   - makedirs: True
+   - user: fluentd
+   - group: fluentd
+{% endfor %}
```

Comment:
This should add the parameter
```
recurse:
  - user
  - group
```

That effectively does a `chown -R user:group directory/path` to ensure proper ownership of subdirectories and files.